starfish/core/image/Filter/max_proj.py

```diff
@@ -55,8 +54,22 @@ def run(
         Returns
         -------
         ImageStack :
-            If in-place is False, return the results of filter as a new stack. Otherwise return the
-            original stack.
+            The max projection of an image across one or more axis.
 
         """
-        return stack.max_proj(*tuple(Axes(dim) for dim in self.dims))
+        max_projection = stack.xarray.max([dim.value for dim in self.dims])
+        max_projection = max_projection.expand_dims(tuple(dim.value for dim in self.dims))
+        max_projection = max_projection.transpose(*stack.xarray.dims)
+        physical_coords: MutableMapping[Coordinates, Sequence[Number]] = {}
+        for axis, coord in (
+                (Axes.X, Coordinates.X),
+                (Axes.Y, Coordinates.Y),
+                (Axes.ZPLANE, Coordinates.Z)):
+            if axis in self.dims:
+                # this axis was projected out of existence.
+                assert coord.value not in max_projection.coords
+                physical_coords[coord] = [np.average(stack.xarray.coords[coord.value])]
+            else:
+                physical_coords[coord] = max_projection.coords[coord.value]
+        max_proj_stack = ImageStack.from_numpy(max_projection.values, coordinates=physical_coords)
+        return max_proj_stack
```

Comment:
wait does it still make sense to have the max_proj filter is stack.reduce('max') does the same thing?

Reply:
max_proj filter is deprecated.  Reduce filter still exists.  stack.reduce is an alias for that filter.